core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -93,9 +93,18 @@ public void run() {
         }
 
         @Override
-        public void onAdvertsError(String message) {
+        public void onAdvertsError(final Exception cause) {
             loadingAds = null;
-            eventListener.onAdLoadError(null, null);
+            advertBreaks = Collections.emptyList();
+            adPlaybackState = new AdPlaybackState();
+
+            handler.post(new Runnable() {
```

Comment:
All callbacks should be performed on the `main-thread` as it is the one from which the player is created.